javatests/arcs/android/integration/policy/PolicyTest.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.android.integration.policy
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.integration.IntegrationEnvironment
+import arcs.core.data.RawEntity
+import arcs.core.host.toRegistration
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.core.util.testutil.LogRule
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalCoroutinesApi::class, kotlin.time.ExperimentalTime::class)
+@RunWith(AndroidJUnit4::class)
+class PolicyTest {
+
+  @get:Rule
+  val log = LogRule()
+
+  /** Note, a new IntegrationEnvironment is allocated for each test method. */
+  @get:Rule
+  val env = IntegrationEnvironment()
+
+  /**
+   * Scenario: A policy-compliant recipe with @persistent handles egresses data
+   * and stores ingress-restricted values to disk that is never deleted.
+   */
+  @Test
+  fun persistentHandlesWithEgress_StoresIngressRestrictedValues_neverDeleted() = runBlocking {
+    env.addNewHostWith(
+      ::IngressThing.toRegistration(),
+      ::EgressAB.toRegistration()
+    )
```

Comment:
I don't think this is testing ingress restricting at all. The persistent fields are `a` and `b` because that is the only thing that is being read.

Reply:
I think this test is exercising ingress restricting capability at the integration level. By checking that `c` and `d` only contain null values in the core assert, we show that this data is never written. 

In an earlier version, I asserted against the `createHandle` arguments. However, Filippo pointed out that this approach was better suited for unit tests, not integration. 

If this is unsatisfactory, do you have any suggestions about alternatives I should take?

Reply:
My comment wasn't related to checking the database for fields "a" and "b".  It was related to the fact that only fields "a" and "b" persist because the recipe only reads those fields. On thinking more about it, this behavior is also considered ingress restricting. So, this is fine.